src/lib/canvas/dropzone.js

```diff
@@ -0,0 +1,98 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+import styled from 'styled-components';
+import { map, filter, throttleTime, distinctUntilChanged } from 'rxjs/operators';
+import { fromEvent } from 'rxjs';
+import { SECTORS, SECTORS_HIGHLIGHT_COLOR, SECTORS_CENTER_RATIO } from './constants';
+import { getSectorCouple, isSmallSize } from './utils';
+
+const Container = styled.div`
+  height: 100%;
+  padding: 0;
+  margin: 0;
+  position:relative;
+`;
+
+const Sector = styled.div`
+  width: 50%;
+  height: 50%;
+  float: left;
+`;
+
+const Centre = styled.div`
+  position: absolute;
+  height: ${props => props.ratio * 100}%;
+  width: ${props => props.ratio * 100}%;
+  top: ${props => (1 - props.ratio) * 50}%;
+  left: ${props => (1 - props.ratio) * 50}%;
+  background: transparent;
+`;
+
+export default class Dropzone extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      sectors: [],
+      ratio: SECTORS_CENTER_RATIO
+    };
+  }
+
+  componentDidMount() {
+    const container = ReactDOM.findDOMNode(this);
+    this.setState({
+      ratio: isSmallSize(container) ? 1 : SECTORS_CENTER_RATIO
+    });
+
+    this.onDrag$ = fromEvent(container, 'dragover').pipe().pipe(
+      map(e => e.preventDefault() || [e.clientX, e.clientY]),
+      distinctUntilChanged((a, b) => a.toString() === b.toString()),
+      throttleTime(50),
+      map(point => document.elementFromPoint(...point).classList.value),
```

Comment:
why not to put the whole mapping logic in one callback instead of three?

Reply:
mmm because it looks better

Reply:
I don't like fat callbacks